src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -215,6 +216,128 @@ def add_imaging_plane(nwbfile: NWBFile, metadata: dict, imaging_plane_index: int
     return nwbfile
 
 
+def add_plane_segmentation(
+    segmentation_extractor: SegmentationExtractor,
+    nwbfile: NWBFile,
+    metadata: Optional[dict],
+    plane_segmentation_index: int = 0,
+    image_segmentation_name: str = None,
+    iterator_options: Optional[dict] = None,
+    compression_options: Optional[dict] = None,
+) -> NWBFile:
+    """
+    Adds the plane segmentation specified by the metadata to the image segmentation.
+    If the plane segmentation already exists in the image segmentation, it is not added again.
+
+    Parameters
+    ----------
+    segmentation_extractor : SegmentationExtractor
+        The segmentation extractor to get the results from.
+    nwbfile : NWBFile
+        The nwbfile to add the plane segmentation to.
+    metadata : dict, optional
+        The metadata for the plane segmentation.
+    plane_segmentation_index: int, optional
+        The index of the plane segmentation to add.
+    image_segmentation_name : str, optional
+        The name of the image segmentation to add the plane segmentation to.
+    iterator_options : dict, optional
+        The options to use when iterating over the image masks of the segmentation extractor.
+    compression_options : dict, optional
+        The options to use when compressing the image masks of the segmentation extractor.
+
+    Returns
+    -------
+    NWBFile
+        The nwbfile passed as an input with the plane segmentation added.
+    """
+    if image_segmentation_name is None:
+        image_segmentation_name = "ImageSegmentation"
+    if iterator_options is None:
+        iterator_options = dict()
+    if compression_options is None:
+        compression_options = dict()
+
+    def image_mask_iterator():
+        for roi_id in segmentation_extractor.get_roi_ids():
+            image_masks = segmentation_extractor.get_roi_image_masks(roi_ids=[roi_id]).T.squeeze()
+            yield image_masks
+
+    # Set the defaults and required infrastructure
+    metadata_copy = deepcopy(metadata)
+    default_metadata = get_default_ophys_metadata()
+    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
+
+    image_segmentation_metadata = metadata_copy["Ophys"]["ImageSegmentation"]
+    plane_segmentation_metadata = image_segmentation_metadata["plane_segmentations"][plane_segmentation_index]
+    plane_segmentation_name = plane_segmentation_metadata["name"]
+
+    # Check if "ophys" processing module already exists in the NWB file
+    if "ophys" not in nwbfile.processing:
```

Comment:
There is a function in `tools.nwb_helpers` called `get_module` which should be (`get_processing_module`). I think this could be used here but I am not sure. Up to you if you want a further code reduction : )

Reply:
Great, wasn't familiar with that one, def useful here!